src/docs/guides/rails.md

```diff
@@ -0,0 +1,200 @@
+        # Can be used by load balancers and uptime monitors to verify that the app is live.
+        get "up" => "rails/health#show", as: :rails_health_check
+
+        # Render dynamic PWA files from app/views/pwa/*
+        get "service-worker" => "rails/pwa#service_worker", as: :pwa_service_worker
+        get "manifest" => "rails/pwa#manifest", as: :pwa_manifest
+
+        # Defines the root path route ("/")
+        root "hello_world#index"
+    end
+    ```
+3. **Modify the View**: Open the `app/views/hello_world/index.html.erb` file and replace its content with the following:
+
+    ```ruby
+    <h1>Hello World</h1>
+
+    <p> This is a Rails app running on Railway</p>
+    ```
+4. **Run the Application Locally**: Start the Rails server by running:
+
+    ```bash
+        bin/rails server
+    ```
+
+    Open your browser and go to `http://localhost:3000` to see your "Hello World" page in action.
+
+Now that your app is running locally, let’s move on to deploying it to Railway!
+
+## Deploy Ruby on Rails App on Railway
+
+Railway offers multiple ways to deploy your Rails app, depending on your setup and preference. Choose any of the following methods:
+
+1. [One-click deploy from a template](#one-click-deploy-from-a-template).
+2. [Using the CLI](#deploy-from-the-cli).
+3. [From a GitHub repository](#deploy-from-a-github-repo).
+
+## One-Click Deploy from a Template
+
+If you’re looking for the fastest way to get started, the one-click deploy option is ideal. It sets up a Rails app along with a Postgres database and Redis.
+
+Click the button below to begin:
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/new/template/sibk1f)
+
+After deploying, we recommend that you [eject from the template](/guides/deploy#eject-from-template-repository) to create a copy of the repository under your own GitHub account. This will give you full control over the source code and project.
+
+## Deploy from the CLI
+
+To deploy the Rails app using the Railway CLI, please follow the steps:
+
+1. **Install the Railway CLI**:
+    - <a href="/guides/cli#installing-the-cli" target="_blank">Install the CLI</a> and <a href="/guides/cli#authenticating-with-the-cli" target="_blank">authenticate it</a> using your Railway account.
+2. **Initialize a Railway Project**:
+    - Run the command below in your Rails app directory. 
+        ```bash
+        railway init
+        ```
+    - Follow the prompts to name your project.
+    - After the project is created, click the provided link to view it in your browser.
+3. **Deploy the Application**:
+    - Use the command below to deploy your app:
+        ```bash
+        railway up
+        ```
+    - This command will scan, compress and upload your app's files to Railway. You’ll see real-time deployment logs in your terminal.
+ - **Note:** If you see an error about a missing `secret_key_base` for the production environment, don’t worry. We’ll fix this in the next step.
```

Comment:
Instead of having the user deploy their app before the database, have them deploy their database before the app so that they can setup the needed variables on the service before initial deploy, this would prevent the unnecessary crash and redeploy.

Reply:
It doesn't matter though. We are deploying via the CLI, even if the database is set up first via the cli, the app service would not exist for the reader to be able to add variables to it before `railway up`.

`railway up` is what creates the service. The only way that it will work is if the readers goes to the project canvas to create an empty service first and then  add the variables. And that entirely defeats the purpose of using the CLI for the first time.

Reply:
Fair point, maybe the CLI should support creating empty services, but that's out of scope.

Reply:
Yes out of scope. 

One thing we can do later is to improve the CLI is add a feature/command to be able to add/set/edit environment variables from the CLI. We currently don't have that option. I used this heavily in [Heroku's CLI ](https://devcenter.heroku.com/articles/config-vars#managing-config-vars)a couple of years back.